src/components/Widgets/Markdown/MarkdownControl/VisualEditor/keys.js

```diff
@@ -20,10 +21,9 @@ function changeHistory(change, type) {
    */
   const next = historyOfType.first();
   const historyOfTypeIsValid = historyOfType.size > 1
-    || next.length > 1
-    || next[0].type !== 'set_selection';
+    || ( next && next.length > 1 && next[0].type !== 'set_selection' );
```

Comment:
This changes the logic - should evaluate true if `next.length > 1` OR `next[0].type !== 'set_selection'`. I also prefer multiline for readability. If there was no bug here, let's just remove the changes and leave as is.

Reply:
Oh I see, you're pulling the `next` check up - you can just prepend it:

Reply:
I changed that because there was a bug, sometimes `next` is `undefined`.
But I don't understand this logic :  `next.length > 1` OR `next[0].type`.
It will throw an error if `next.length === 0`.

Reply:
I actually pulled this logic from Slate's core - we could make it safe by changing the last line to: `|| (next[0] && next[0].type !== 'set_selection');`